block/manager.go

```diff
@@ -736,14 +735,21 @@ func (m *Manager) createNextDABatch(startHeight uint64, endHeight uint64) (*type
 			return nil, err
 		}
 
+		batch.Blocks = append(batch.Blocks, block)
+		batch.Commits = append(batch.Commits, commit)
+
 		//Check if the batch size is too big
-		totalSize = totalSize + block.ToProto().Size() + commit.ToProto().Size()
-		if totalSize >= int(m.conf.BlockBatchSizeBytes) {
+		totalSize := batch.ToProto().Size()
+		if m.conf.BlockBatchSizeBytes > 0 && totalSize >= int(m.conf.BlockBatchSizeBytes) {
+			// Nil out the last block and commit
+			batch.Blocks[len(batch.Blocks)-1] = nil
```

Comment:
Is setting it to nil it before removing it is just for the garbage collector or is there any other reason?

Reply:
yes, probably not mandatory, but to be on the safe side (as the array is of pointers)
otherwise, the "deleted" elements will still exist as long as the "batch" struct exists